src/hotspot/share/gc/shenandoah/shenandoahEvacTracker.hpp

```diff
@@ -97,7 +100,11 @@ class ShenandoahEvacuationTracker : public CHeapObj<mtGC> {
 public:
   ShenandoahEvacuationTracker() = default;
 
+  // Record that the given thread has begun to evacuate an object of this size.
   void begin_evacuation(Thread* thread, size_t bytes, ShenandoahAffiliation from, ShenandoahAffiliation to);
+
+  // Multiple threads may attempt to evacuate the same object, but only the succeeding thread will end the evacuation.
```

Comment:
Instead of `only the succeeding thread will end the evacuation`, I might say: `the unique thread that is successful will end the evacuation`. (Alternatively, replace "succeeding" with "successful"; the word "succeeding" as an adjective can be ambiguous in this context.)